live_pipeline.py

```diff
@@ -0,0 +1,74 @@
+from os import getenv
+import json
+from datetime import datetime, date
+import boto3
+from botocore import client
+from extract import get_listing_data, get_max_page_num
+from transform import get_data, assemble_data
+from load import get_connection, insert_to_database
+
+FILE_NAME = "log.json"
+BUCKET_NAME = "c12-court-transcripts"
+
+
+def get_client() -> client:
+    """Initiates a connection to the S3 AWS Cloud using required credentials."""
+    aws_client = boto3.client(
+        "s3",
+        aws_access_key_id=getenv("ACCESS_KEY_ID"),
+        aws_secret_access_key=getenv("SECRET_ACCESS_KEY"),
+    )
+    return aws_client
+
+
+def read_from_json(client: client) -> dict:
+    """Retrieves record of cases already processed from json file on S3."""
+    client.download_file(BUCKET_NAME, FILE_NAME, FILE_NAME)
+    with open(FILE_NAME, "r", encoding="utf-8") as f:
+        log_json = json.load(f)
+    return log_json
+
+
+def handler(event: dict, context) -> None:
+    client = get_client()
+    count = 0
+    log_json = read_from_json(get_client())
+    log = list(log_json.values())[0]
+    log_date = list(log_json.keys())[0]
+    day, month, year = log_date.split("-")
+    LIVE_URL = f"""https://caselaw.nationalarchives.gov.uk/judgments/search?per_page=50&order=date&query=&from_date_0={day}&from_date_1={month}&from_date_2={year}&to_date_0=&to_date_1=&to_date_2=&court=uksc&court=ukpc&court=ewca%2Fciv&court=ewca%2Fcrim&court=ewhc%2Fadmin&court=ewhc%2Fadmlty&court=ewhc%2Fch&court=ewhc%2Fcomm&court=ewhc%2Ffam&court=ewhc%2Fipec&court=ewhc%2Fkb&court=ewhc%2Fmercantile&court=ewhc%2Fpat&court=ewhc%2Fscco&court=ewhc%2Ftcc&party=&judge=&page="""
```

Comment:
Should LIVE_URL be outside the function?

Reply:
In this case, it needed to be inside as it is changing the URL in places using fstrings. The alternative could be to put it outside the function as a global and use regex statements to manipulate the URL.